### DISCUSSION TITLE
Error handling

### DISCUSSION THREAD

**Discussion by andriyndev** [serverpod/serverpod#826] [Q&A] [answered]
What is the correct way to handle exceptions that might occur when calling endpoints? I can catch all the exceptions with try/catch, but  I didn't find in the docs anything about how to handle a specific exception.

**Comment by yahu1031** [comment]
These are the possible exceptions you get 
- `SerializableException`
- `ServerpodClientException`
- `S3Exception`
- `NoPermissionsException`
- `UserImageException`
- `CloudStorageException`
- `PostgreSQLException`
- `ResultInternalServerError`

You can handle it normally using try/catch. And errors from the server will be passed to the client, so it will be much flexible to handle now.

**Reply by andriyndev** [reply]
Thank you! I think, it'd be great to describe them in the documentation because it may be complicated to find all of them in the source code. And the topic is important because the user will get frustrated if they doesn't know why the app doesn't work as expected and who is responsible for it.

**Reply by yahu1031** [reply]
Yes, Agree with this. I think @vlidholt is on his way to this already.

**Comment by vlidholt** [comment, marked as answer]
Serverpod 1.1 was soft launched today. Among many other things it adds support for better error handling. Read more here: https://docs.serverpod.dev/concepts/exceptions
